Fix macro redefined warnings from clang.h when using g++ / clang++ on Windows

There are a set of preprocessor macros defined for both the Windows platform and for the GCC compiler.  This results in the following errors for example when using g++ on Windows with the slang.h header:

```
../../slang/include/slang.h:277:13: warning: "SLANG_NO_INLINE" redefined
  277 |     #define SLANG_NO_INLINE __declspec(noinline)
      |             ^~~~~~~~~~~~~~~
../../slang/include/slang.h:260:13: note: this is the location of the previous definition
  260 |     #define SLANG_NO_INLINE __attribute__((noinline))
      |             ^~~~~~~~~~~~~~~
../../slang/include/slang.h:278:13: warning: "SLANG_FORCE_INLINE" redefined
  278 |     #define SLANG_FORCE_INLINE __forceinline
      |             ^~~~~~~~~~~~~~~~~~
../../slang/include/slang.h:261:13: note: this is the location of the previous definition
  261 |     #define SLANG_FORCE_INLINE inline __attribute__((always_inline))
      |             ^~~~~~~~~~~~~~~~~~
../../slang/include/slang.h:279:13: warning: "SLANG_BREAKPOINT" redefined
  279 |     #define SLANG_BREAKPOINT(id) __debugbreak();
      |             ^~~~~~~~~~~~~~~~
../../slang/include/slang.h:262:13: note: this is the location of the previous definition
  262 |     #define SLANG_BREAKPOINT(id) __builtin_trap();
      |             ^~~~~~~~~~~~~~~~
../../slang/include/slang.h:280:13: warning: "SLANG_ALIGN_OF" redefined
  280 |     #define SLANG_ALIGN_OF(T) __alignof(T)
      |             ^~~~~~~~~~~~~~
../../slang/include/slang.h:263:13: note: this is the location of the previous definition
  263 |     #define SLANG_ALIGN_OF(T) __alignof__(T)
      |             ^~~~~~~~~~~~~~
```

--- a/include/slang.h
+++ b/include/slang.h
@@ -273,7 +273,7 @@ convention for interface methods.
 #endif
 
 // Microsoft VC specific
-#if SLANG_MICROSOFT_FAMILY
+#if SLANG_VC
     #define SLANG_NO_INLINE __declspec(noinline)
     #define SLANG_FORCE_INLINE __forceinline
     #define SLANG_BREAKPOINT(id) __debugbreak();

